<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Analyses – pbxplore.analysis &mdash; PBxplore API 1.1 documentation</title>
    
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '1.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="PBxplore API 1.1 documentation" href="../index.html" />
    <link rel="next" title="Directly accessible functions" href="pbxplore.html" />
    <link rel="prev" title="Input/Output in files — pbxplore.io" href="io.html" />
   
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9">

  </head>
  <body role="document">  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <span class="target" id="module-pbxplore.analysis"></span><div class="section" id="analyses-pbxplore-analysis">
<h1>Analyses &#8211; <a class="reference internal" href="#module-pbxplore.analysis" title="pbxplore.analysis"><code class="xref py py-mod docutils literal"><span class="pre">pbxplore.analysis</span></code></a><a class="headerlink" href="#analyses-pbxplore-analysis" title="Permalink to this headline">¶</a></h1>
<div class="section" id="build-occurence-matrix">
<h2>Build occurence matrix<a class="headerlink" href="#build-occurence-matrix" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pbxplore.analysis.count_matrix">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">count_matrix</code><span class="sig-paren">(</span><em>pb_seq</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/count.html#count_matrix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.count_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Count the occurences of each block at each position.</p>
<p>The occurence matrix has one row per sequence, and one column per block.
The columns are ordered in as PB.NAMES.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>pb_seq</strong> (<em>a list of PB sequences.</em>) &#8211; </td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"></td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">The occurence matrix.</td>
</tr>
<tr class="field-even field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal"><span class="pre">InvalidBlockError</span></code> &#8211; encountered an unexpected PB</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pbxplore.analysis.read_occurence_file">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">read_occurence_file</code><span class="sig-paren">(</span><em>name</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/count.html#read_occurence_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.read_occurence_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Read an occurence matrix from a file.
It will return the matrix as a numpy array and the indexes of residues.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>name</strong> (<em>str</em>) &#8211; Name of the file.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><ul class="simple">
<li><strong>count_mat</strong> (<em>numpy array</em>) &#8211; the occurence matrix without the residue number</li>
<li><strong>residues</strong> (<em>list</em>) &#8211; the list of residues indexes</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal"><span class="pre">ValueError</span></code> &#8211; when something is wrong about the file</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pbxplore.analysis.plot_map">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">plot_map</code><span class="sig-paren">(</span><em>fname</em>, <em>count_mat</em>, <em>residue_min=1</em>, <em>residue_max=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/visualization.html#plot_map"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.plot_map" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a map of the distribution of PBs along protein sequence from
an occurence matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fname</strong> (<em>str</em>) &#8211; The path to the file to write in</li>
<li><strong>count_mat</strong> (<em>numpy array</em>) &#8211; an occurence matrix returned by <cite>count_matrix</cite>.</li>
<li><strong>residue_min</strong> (<em>int</em>) &#8211; the lower bound of the protein sequence</li>
<li><strong>residue_max</strong> (<em>int</em>) &#8211; the upper bound of the protein sequence</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="compare-protein-block-sequences">
<h2>Compare protein block sequences<a class="headerlink" href="#compare-protein-block-sequences" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pbxplore.analysis.compare">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">compare</code><span class="sig-paren">(</span><em>header_lst</em>, <em>seq_lst</em>, <em>substitution_mat</em>, <em>fname</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/compare.html#compare"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.compare" title="Permalink to this definition">¶</a></dt>
<dd><p>Command line wrapper for the comparison of all sequences with the first one</p>
<p>When the &#8211;compare option is given to the command line, the program
compares all the sequences to the first one and writes these comparison as
sequences of digits. These digits represent the distance between the PB
in the target and the one in the reference at the same position. The digits
are normalized in the [0; 9] range.</p>
<p>This function run the comparison, write the result in a fasta file, and
display on screen informations about the process.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>header_lst</strong> (<em>list of strings</em>) &#8211; The list of sequence headers ordered as the sequences</li>
<li><strong>seq_lst</strong> (<em>list of strings</em>) &#8211; The list of sequences ordered as the headers</li>
<li><strong>substitution_mat</strong> (<em>numpy.array</em>) &#8211; A substitution matrix expressed as similarity scores</li>
<li><strong>fname</strong> (<em>str</em>) &#8211; The output file name</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="visualize-deformability">
<h2>Visualize deformability<a class="headerlink" href="#visualize-deformability" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pbxplore.analysis.compute_neq">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">compute_neq</code><span class="sig-paren">(</span><em>count_mat</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/neq.html#compute_neq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.compute_neq" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the Neq for each residue from an occurence matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>count_mat</strong> (<em>numpy array</em>) &#8211; an occurence matrix returned by <cite>count_matrix</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>neq_array</strong> &#8211; a 1D array containing the neq values</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">numpy array</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pbxplore.analysis.plot_neq">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">plot_neq</code><span class="sig-paren">(</span><em>fname</em>, <em>neq_array</em>, <em>residue_min=1</em>, <em>residue_max=None</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/visualization.html#plot_neq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.plot_neq" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate the Neq plot along the protein sequence</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>neq_array</strong> (<em>numpy array</em>) &#8211; an array containing the neq value associated to the residue number</li>
<li><strong>fname</strong> (<em>str</em>) &#8211; The path to the file to write in</li>
<li><strong>residue_min</strong> (<em>int</em>) &#8211; the lower bound of the protein sequence</li>
<li><strong>residue_max</strong> (<em>int</em>) &#8211; the upper bound of the protein sequence</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pbxplore.analysis.generate_weblogo">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">generate_weblogo</code><span class="sig-paren">(</span><em>fname</em>, <em>count_mat</em>, <em>residue_min=1</em>, <em>residue_max=None</em>, <em>title=''</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/visualization.html#generate_weblogo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.generate_weblogo" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates logo representation of PBs frequency along protein sequence through
the weblogo library.</p>
<p>The weblogo reference:
G. E. Crooks, G. Hon, J.-M. Chandonia, and S. E. Brenner.
&#8216;WebLogo: A Sequence Logo Generator.&#8217;
Genome Research 14:1188–90 (2004)
doi:10.1101/gr.849004.
http://weblogo.threeplusone.com/</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fname</strong> (<em>str</em>) &#8211; The path to the file to write in</li>
<li><strong>count_mat</strong> (<em>numpy array</em>) &#8211; an occurence matrix returned by <cite>count_matrix</cite>.</li>
<li><strong>residue_min</strong> (<em>int</em>) &#8211; the lower bound of residue frame</li>
<li><strong>residue_max</strong> (<em>int</em>) &#8211; the upper bound of residue frame</li>
<li><strong>title</strong> (<em>str</em>) &#8211; the title of the weblogo. Default is empty.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="cluster-protein-block-sequences">
<h2>Cluster protein block sequences<a class="headerlink" href="#cluster-protein-block-sequences" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="utils">
<h2>Utils<a class="headerlink" href="#utils" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pbxplore.analysis.substitution_score">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">substitution_score</code><span class="sig-paren">(</span><em>substitution_matrix</em>, <em>seqA</em>, <em>seqB</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/utils.html#substitution_score"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.substitution_score" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the substitution score to go from <cite>seqA</cite> to <cite>seqB</cite></p>
<p>Both sequences must have the same length.</p>
<p>The score is either expressed as a similarity or a distance depending on
the substitution matrix.</p>
</dd></dl>

<dl class="function">
<dt id="pbxplore.analysis.compute_freq_matrix">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">compute_freq_matrix</code><span class="sig-paren">(</span><em>count_mat</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/utils.html#compute_freq_matrix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.compute_freq_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute a PB frequency matrix from an occurence matrix.</p>
<p>The frequency matrix has one row per sequence, and one column per block.
The columns are ordered in as PB.NAMES.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>count_mat</strong> (<em>numpy array</em>) &#8211; an occurence matrix returned by <cite>count_matrix</cite>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><strong>freq</strong> &#8211; The frequency matrix</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">numpy array</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pbxplore.analysis.compute_score_by_position">
<code class="descclassname">pbxplore.analysis.</code><code class="descname">compute_score_by_position</code><span class="sig-paren">(</span><em>score_mat</em>, <em>seq1</em>, <em>seq2</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pbxplore/analysis/utils.html#compute_score_by_position"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pbxplore.analysis.compute_score_by_position" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes substitution score between two sequences position per position</p>
<p>The substitution score can represent a similarity or a distance depending
on the score matrix provided. The score matrix should be provided as a 2D
numpy array with score[i, j] the score to swich the PB at the i-th position
in PB.NAMES to the PB at the j-th position in PB.NAMES.</p>
<p>The function returns the result as a list of substitution scores to go from
<cite>seq1</cite> to <cite>seq2</cite> for each position. Both sequences must have the same
length.</p>
<p>..note:</p>
<blockquote>
<div>The score to move from or to a Z block (dummy block) is always 0.</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Raises:</th><td class="field-body"><code class="xref py py-exc docutils literal"><span class="pre">InvalidBlockError</span></code> &#8211; encountered an unexpected PB</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Analyses &#8211; <code class="docutils literal"><span class="pre">pbxplore.analysis</span></code></a><ul>
<li><a class="reference internal" href="#build-occurence-matrix">Build occurence matrix</a></li>
<li><a class="reference internal" href="#compare-protein-block-sequences">Compare protein block sequences</a></li>
<li><a class="reference internal" href="#visualize-deformability">Visualize deformability</a></li>
<li><a class="reference internal" href="#cluster-protein-block-sequences">Cluster protein block sequences</a></li>
<li><a class="reference internal" href="#utils">Utils</a></li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="io.html" title="previous chapter">Input/Output in files &#8212; <code class="docutils literal"><span class="pre">pbxplore.io</span></code></a></li>
      <li>Next: <a href="pbxplore.html" title="next chapter">Directly accessible functions</a></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/pages/analysis.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2015, Pierre Poulain, and Alexandre G. deBrevern.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.3.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.6</a>
      
      |
      <a href="../_sources/pages/analysis.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>